pulp_smash/tests/puppet/cli/test_sync_from_puppet_skip_duplicate_content.py

```diff
@@ -0,0 +1,102 @@
+    into two repositories and check whether the 2nd would have same amount
+    of content units with the 1st.
+
+    .. _Pulp #1937: https://pulp.plan.io/issues/1937
+
+    """
+
+    def test_sync_puppet_forge(self):
+        """Create two Puppet repositories and trigger their syncs."""
+        cfg = config.get_config()
+        utils.pulp_admin_login(cfg)
+
+        # Create two different repositories and schedule them for syncs.
+        repo_ids = [utils.uuid4() for _ in range(2)]
+        client = cli.Client(cfg)
+
+        for repo_id in repo_ids:
+            # Create the repository.
+            cmd = (
+                'pulp-admin puppet repo create '
+                '--repo-id {} --feed {} --queries {}'
+            ).format(repo_id, PUPPET_FEED, PUPPET_QUERY)
+            client.run(cmd.split())
+
+            # Delete the repository.
+            cmd = 'pulp-admin puppet repo delete --repo-id {}'.format(repo_id)
+            self.addCleanup(client.run, cmd.split())
+
+            # Sync the repository.
+            cmd = (
+                'pulp-admin puppet repo sync run --repo-id {}'
+            ).format(repo_id)
+            client.run(cmd.split())
+
+        # Verify that all the repositories have an identical non-zero
+        # number of content units present with the 1st one
+        first_repo_counts = self.get_conent_counts_by_repo_id(
+            repo_ids[0],
+            client
+        )
+        self.assertIsNot(first_repo_counts, 0, 'Counts should be non-zero.')
+        for repo_id in repo_ids:
+            # Compute current repository's content unit counts
+            if repo_id is not repo_ids[0]:
+                counts = self.get_conent_counts_by_repo_id(repo_id, client)
+                self.assertIsNot(counts, 0, 'Counts should be non-zero.')
+                self.assertEqual(
+                    first_repo_counts,
+                    counts,
+                    'The numbers of content units are not equal.'
+                )
+
+    @staticmethod
+    def get_conent_counts_by_repo_id(repo_id, client):
+        """Extract number of content unit counts by repository ID.
+
+        Given the same feed URL, the 1st synced repo would have non-zero
+        number of content units. However, the remaining repos might have
+        0 counts, which is actually not correct.
+
+        Please refer to the issue comment:
+        https://github.com/PulpQE/pulp-smash/issues/269#issuecomment-224427393
+
+        :param repo_id: the repository's id with which to get content counts
+        :param  client: the convenience object to execute commands
+        :return:    the number of content units of repository
+        """
+        cmd = (
+            'pulp-admin puppet repo list --repo-id {} '
+            '--fields content_unit_counts'
+        ).format(repo_id)
+        completed_proc = client.run(cmd.split())
+        output = getattr(completed_proc, 'stdout').split('Puppet Module:')
```

Comment:
What happens if there's more than one type of content unit in this repository? For example:

```python
>>> mytext = (
...   'foo bar biz baz\n'
...   'Puppet Module: 5\n'
...   'one two three four'
... )
>>> mytext.split('Puppet Module:')
['foo bar biz baz\n', ' 5\none two three four']
```

In this case, the call to `int(output[1].strip())` will fail. A more robust solution is to find the one line of text containing the keyword 'Puppet Module:', and to parse that one line. Also, this logic seems like something that other test cases might want to make use of. A standalone function will be more re-usable.

Use a function like the following instead:

```python
def _get_n_puppet_modules(server_config, repo_id):
    """Tell how many puppet modules are in a repository.

    :param pulp_smash.config.ServerConfig server_config: Information about the
        Pulp server being targeted.
    :param repo_id: A Puppet repository ID.
    :returns: The number of puppet modules in a repository, as an ``int``.
    """
    keyword = 'Puppet Module:'
    completed_proc = cli.Client(server_config).run((
        'pulp-admin puppet repo list --repo-id {} --fields content_unit_counts'
    ).format(repo_id).split())
    lines = [
        line for line in completed_proc.stdout.splitlines() if keyword in line
    ]
    assert len(lines) < 2
    if len(lines) == 0:
        return 0
    else:
        return int(lines[1].split(keyword)[1].strip())
```